src/abstractmcmc.jl

```diff
@@ -303,3 +239,138 @@ function (cb::HMCProgressCallback)(rng, model, spl, t, state, i; nadapts = 0, kw
         @info "Finished $nadapts adapation steps" adaptor κ.τ.integrator metric
     end
 end
+
+#############
+### Utils ###
+#############
+
+function get_type_of_spl(spl::AbstractHMCSampler)
+    T = collect(typeof(spl).parameters)[1]
+    return T
+end
```

Comment:
This seems too hacky for my taste. According to a quick check, (unsurprisingly) it also can't be inferred. It seems this could be fixed by removing `collect` - but IMO the correct approach would be to define dispatches such as `get_type_of_spl(::NUTS{T}) = T` - or more generically define `abstract type AbstractHMCSampler{T<:Real} <: AbstractSampler end` and define `NUTS{T<:Real} <: AbstractHMCSampler{T} ...`.

Reply:
Thanks David! I was trying to get something like this to work for 1h yesterday until I exasperated.